v4/newrelic/transaction.go

```diff
@@ -125,7 +181,10 @@ func (txn *Transaction) StartSegmentNow() SegmentStartTime {
 //	// ... code you want to time here ...
 //	segment.End()
 func (txn *Transaction) StartSegment(name string) *Segment {
-	return nil
+	return &Segment{
+		StartTime: txn.StartSegmentNow(),
+		Name:      name,
```

Comment:
Is there any reason to store the name instead of passing it right to `tracer.Start` via `StartSegmentNow`?

Reply:
Yeah, good question. I did it this way because a customer could do something like this:

```go
seg := &Segment{
    StartTime: txn.StartSegmentNow(),
}
seg.Name = "hello"
seg.End()
```

In this example, they have set the name _after_ they have started it.